### PR TITLE
feat: edit date-picker to hook

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    eidt `date-picker` to hook     |
| 🇨🇳 Chinese |    把`date-picker `   改成 hook    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 1]
```diff
@@ -21,137 +21,103 @@ import { PickerComponentClass } from './interface';
...
-                {size => {
-                  const mergedSize = customizeSize || size;
+    const suffixNode = (
+      <>
+        {picker === 'time' ? <ClockCircleOutlined /> : <CalendarOutlined />}
+        {hasFeedback && feedbackIcon}
+      </>
+    );
+    return (
+      <LocaleReceiver componentName="DatePicker" defaultLocale={enUS}>
```
Comment (heiyu4585) [inline components/date-picker/generatePicker/generateRangePicker.tsx:72]: 这个应该不用改成 useContext 吧

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 2]
```diff
@@ -21,137 +21,103 @@ import { PickerComponentClass } from './interface';
...
+      disabled: customDisabled,
+      bordered = true,
+      placeholder,
+      status: customStatus,
+      ...restProps
+    } = props;
 
-    pickerRef = React.createRef<RCRangePicker<DateType>>();
-
-    focus = () => {
```
Comment (heiyu4585) [inline components/date-picker/generatePicker/generateRangePicker.tsx:31]: focus ,blur 我直接删除了,是否可以

File: components/time-picker/__tests__/index.test.js [hunk 3]
```diff
@@ -62,26 +63,34 @@ describe('TimePicker', () => {
...
 
-  it('should pass popupClassName prop to Picker as dropdownClassName prop', () => {
-    const popupClassName = 'myCustomClassName';
+  it('should pass dropdownClassName prop to Picker', () => {
+    const dropdownClassName = 'myCustomClassName';
     const wrapper = mount(
       <TimePicker
         defaultOpenValue={moment('00:00:00', 'HH:mm:ss')}
-        popupClassName={popupClassName}
+        dropdownClassName={dropdownClassName}
```
Comment (zombieJ) [inline components/time-picker/__tests__/index.test.js:71]: 不能改名字，这样 break 现有的用户了。

File: components/date-picker/generatePicker/generateRangePicker.tsx (hunk 1 above)
Comment (heiyu4585) [reply components/date-picker/generatePicker/generateRangePicker.tsx:72]: @zombieJ

Comment (zombieJ) [reply components/date-picker/generatePicker/generateRangePicker.tsx:72]: 不用，这个 component 自带逻辑了。要改的话是做成 hooks，这个不用在这个 PR 里做

File: components/date-picker/generatePicker/generateRangePicker.tsx (hunk 2 above)
Comment (zombieJ) [reply components/date-picker/generatePicker/generateRangePicker.tsx:31]: 加个测试用例来测一测，应该是 OK 的

Comment (MadCcc) [reply components/date-picker/generatePicker/generateRangePicker.tsx:31]: 原有的最好还是在 ref 里提供下吧

File: components/time-picker/__tests__/index.test.js (hunk 3 above)
Comment (heiyu4585) [reply components/time-picker/__tests__/index.test.js]: done

File: components/date-picker/generatePicker/generateRangePicker.tsx (hunk 2 above)
Comment (heiyu4585) [reply components/date-picker/generatePicker/generateRangePicker.tsx:31]: em...done

File: components/date-picker/generatePicker/generateRangePicker.tsx [hunk 4]
```diff
@@ -55,105 +36,108 @@ export default function generateRangePicker<DateType>(
...
+      const suffixNode = (
+        <>
+          {picker === 'time' ? <ClockCircleOutlined /> : <CalendarOutlined />}
+          {hasFeedback && feedbackIcon}
+        </>
       );
-    };
 
-    render() {
+      const focus = () => {
```
Comment (MadCcc) [inline components/date-picker/generatePicker/generateRangePicker.tsx:76]: 用 `useImperativeHandle` 包含 focus，blur 和 pickerRef，尽量和 class component 拿到的 ref 功能上保持一致

**Review by zombieJ — APPROVED** [review]

**Pull request merged**